src/hooks/outsideClickHook.js

```diff
@@ -13,5 +13,3 @@ const useOutsideClickHook = (ref, callback) => {
     };
   }, [ref, callback]);
```

Comment:
your callback is always new on each rerender, so on each rerender you add and remove event listeners.
improve code, so you only add event listener once